src/Tasks/Microsoft.NET.Build.Tasks/ResolveReadyToRunCompilers.cs

```diff
@@ -181,42 +181,19 @@ private bool GetCrossgen2TargetOS(out string targetOS)
 
             // Determine targetOS based on target rid.
             // Use the runtime graph to support non-portable target rids.
+            // Use the full target rid instead of just the target OS as the runtime graph
+            // may only have the full target rid and not an OS-only rid for non-portable target rids
+            // added by our source-build partners.
             var runtimeGraph = new RuntimeGraphCache(this).GetRuntimeGraph(RuntimeGraphPath);
             string portablePlatform = NuGetUtils.GetBestMatchingRid(
                     runtimeGraph,
-                    _targetPlatform,
-                    new[] { "linux", "linux-musl", "osx", "win", "freebsd", "illumos" },
+                    _targetRuntimeIdentifier,
+                    new[] { "linux", "osx", "win", "freebsd", "illumos" },
                     out _);
 
-            // For source-build, allow the bootstrap SDK rid to be unknown to the runtime repo graph.
```

Comment:
This code was added because we ran into situations **during the runtime repo build** where the runtime graph used in the lookup above did not know the non-portable RID yet.

For the final produced non-portable SDK, the lookup above will works (because the runtime graph that ships with the SDK includes the non-portable rid).

Reply:
As long as we flow the RID graph within the runtime repo (which I think we do but I'm not 100% positive), we don't need this any more. We have changes in ASP.NET Core to flow the RID graph and this block isn't needed when we flow it.

Reply:
@jkoritzinsky whether this can be left out depends on the ordering of things in the runtime repo build. If it is still needed, we can always add it back.

Reply:
We also have custom versions of the tasks in the runtime repo, so we can use those in those scenarios if necessary.